OctoAwesome/OctoAwesome.Client/Components/EntityComponent.cs

```diff
@@ -87,10 +85,92 @@ public void Draw(Matrix view, Matrix projection, Index3 chunkOffset, Index2 plan
                     Matrix world = Matrix.CreateTranslation(
                         shift.X * Chunk.CHUNKSIZE_X + position.LocalPosition.X,
                         shift.Y * Chunk.CHUNKSIZE_Y + position.LocalPosition.Y,
-                        shift.Z * Chunk.CHUNKSIZE_Z + position.LocalPosition.Z) * Matrix.CreateScaling(body.Radius * 2, body.Radius * 2, body.Height)*Matrix.CreateRotationZ(rotation);
-                    effect.World = world;
+                        shift.Z * Chunk.CHUNKSIZE_Z + position.LocalPosition.Z) 
+                                   * Matrix.CreateScaling(body.Radius /40, body.Radius /40, body.Height/80)
+                                   *Matrix.CreateRotationZ(rotation);
+                    
+                    effect.Parameters["WorldViewProj"].SetValue(viewProjection*world);
+                    
                     modelinfo.model.Transform = world;
-                    modelinfo.model.Draw(effect, modelinfo.texture);
+                    modelinfo.model.Draw();
+                }
+            }
+        }
+        
+        public void Draw(Matrix view, Matrix projection, Matrix shadowViewProjection,RenderTarget2D shadowMap , Index3 chunkOffset, Index2 planetSize)
+        {
+            graphicsDevice.RasterizerState = RasterizerState.CullClockwise;
+            foreach (var pass in effect.Techniques["EntityBasic"].Passes)
+            {
+                pass.Apply();
+
+                foreach (var entity in Entities)
+                {
+                    if (!entity.Components.ContainsComponent<RenderComponent>())
+                    {
+                        continue;
+                    }
+
+                    var rendercomp = entity.Components.GetComponent<RenderComponent>();
+
+                    ModelInfo modelinfo;
+
+                    if (!models.TryGetValue(rendercomp.Name,out modelinfo))
+                    {
+                        modelinfo = new ModelInfo()
+                        {
+                            render = true,
+                            model = Game.Content.Load<Model>(rendercomp.ModelName),
+                            texture = Game.Content.Load<Texture2D>(rendercomp.TextureName),
+                    };
```

Comment:
Formatierung beachten, vllt. mal strg + K + D bemühen oder eben das gegenstück in Rider :D